In [26]:
import pandas as pd

from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split


In [27]:
df = pd.read_csv("possum.csv")

In [28]:
df.info();


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 104 entries, 0 to 103
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   case      104 non-null    int64  
 1   site      104 non-null    int64  
 2   Pop       104 non-null    object 
 3   sex       104 non-null    object 
 4   age       102 non-null    float64
 5   hdlngth   104 non-null    float64
 6   skullw    104 non-null    float64
 7   totlngth  104 non-null    float64
 8   taill     104 non-null    float64
 9   footlgth  103 non-null    float64
 10  earconch  104 non-null    float64
 11  eye       104 non-null    float64
 12  chest     104 non-null    float64
 13  belly     104 non-null    float64
dtypes: float64(10), int64(2), object(2)
memory usage: 11.5+ KB


In [29]:
X = df[["skullw", "belly"]]
y = df[["taill"]]

In [30]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [31]:
model = LinearRegression()
model.fit(X_train, y_train)

LinearRegression()

In [32]:
slope = model.coef_[0]  #These are x
intercept = model.intercept_  #This is y

print(slope, intercept)

[0.11254037 0.14364749] [25.97884736]


In [33]:
# y = 25.98 + 0.11x + 0.14x

In [34]:
from sklearn.metrics import mean_squared_error, r2_score

In [35]:
import math

In [36]:
y_pred = model.predict(X_test)
mse = math.sqrt(mean_squared_error(y_test, y_pred))
r2 = r2_score(y_test, y_pred)
print(mse, r2)

1.8582006945008498 0.093071333484123


In [37]:
tailLength = 25.98 + ((0.11 * 58.2) + (0.14 * 34))

In [38]:
tailLength

37.142

#### The tail length for a possum with a skull width of 58.2 cm and a belly girth of 34 cm is 37.1 cm!

In [39]:
unreasonablyLongList = model.predict(X_train)

In [40]:
df["taill"].max()

43.0